apps/api/src/routers/admin.py

```diff
@@ -149,28 +152,49 @@ async def applicant_summary() -> dict[ApplicantStatus, int]:
 
 @router.post("/review")
 async def submit_review(
-    applicant: str = Body(),
-    decision: Decision = Body(),
+    applicant_review: ReviewRequestModel,
     reviewer: User = Depends(require_role({Role.REVIEWER})),
 ) -> None:
-    """Submit a review decision from the reviewer for the given applicant."""
-    log.info("%s reviewed applicant %s", reviewer, applicant)
+    """Submit a review decision from the reviewer for the given hacker applicant."""
+    log.info("%s reviewed hacker %s", reviewer, applicant_review.applicant)
 
-    review: OtherReview = (utc_now(), reviewer.uid, decision)
+    review: Review = (utc_now(), reviewer.uid, applicant_review.score)
 
-    try:
-        await mongodb_handler.raw_update_one(
-            Collection.USERS,
-            {"_id": applicant},
-            {
-                "$push": {"application_data.reviews": review},
-                "$set": {"status": "REVIEWED"},
-            },
-        )
-    except RuntimeError:
-        log.error("Could not submit review for %s", applicant)
+    await _try_update_applicant_with_query(
+        applicant_review,
+        query={"$push": {"application_data.reviews": review}},
+        err_msg=f"{reviewer} could not submit review for {applicant_review.applicant}",
+    )
+
+    applicant_record = await mongodb_handler.retrieve_one(
+        Collection.USERS,
+        {"_id": applicant_review.applicant},
+        ["_id", "application_data.reviews", "roles"],
+    )
+    if not applicant_record:
+        log.error("Could not retrieve applicant after submitting review")
         raise HTTPException(status.HTTP_500_INTERNAL_SERVER_ERROR)
 
+    if Role.HACKER in applicant_record["roles"]:
+        num_reviewers = applicant_review_processor.get_num_unique_reviewers(
+            applicant_record
+        )
```

Comment:
Thought: I'm not entirely sure how [MongoDB's casual consistency](https://www.mongodb.com/docs/manual/core/causal-consistency-read-write-concerns/) works, but there may be an edge case where the previously written data is not immediately available at the next read. Will need to look into this more.